utils/cobrautil/templates/yaml_flag_printer.go

```diff
@@ -0,0 +1,69 @@
+// Copyright 2023 Sauce Labs Inc. All rights reserved.
+//
+// This Source Code Form is subject to the terms of the Mozilla Public
+// License, v. 2.0. If a copy of the MPL was not distributed with this
+// file, You can obtain one at https://mozilla.org/MPL/2.0/.
+
+package templates
+
+import (
+	"bytes"
+	"fmt"
+	"io"
+	"strings"
+
+	"github.com/mitchellh/go-wordwrap"
+	"github.com/spf13/pflag"
+)
+
+type YamlFlagPrinter struct {
+	out       io.Writer
+	wrapLimit uint
+}
+
+func NewYamlFlagPrinter(out io.Writer, wrapLimit uint) *YamlFlagPrinter {
+	return &YamlFlagPrinter{
+		out:       out,
+		wrapLimit: wrapLimit,
+	}
+}
+
+func (p *YamlFlagPrinter) PrintHelpFlag(f *pflag.Flag) {
+	formatBuf := new(bytes.Buffer)
+	writeYamlFlag(formatBuf, f)
+
+	wrappedStr := formatBuf.String()
+	flagAndUsage := strings.Split(formatBuf.String(), "\n")
+	flagStr := flagAndUsage[len(flagAndUsage)-1]
+
+	// if the flag usage is longer than one line, wrap it again
+	if len(flagAndUsage) > 1 {
+		nextLines := strings.Join(flagAndUsage[:len(flagAndUsage)-1], " ")
+		wrappedUsages := wordwrap.WrapString(nextLines, p.wrapLimit-2)
+		wrappedUsages = "# " + strings.ReplaceAll(wrappedUsages, "\n", "\n# ")
+		wrappedStr = wrappedUsages + "\n#" + flagStr
+	}
+	fmt.Fprintf(p.out, wrappedStr)
+	fmt.Fprintf(p.out, "\n\n")
+}
+
+// writeFlag will output the help flag based
```

Comment:
outdated comment - function name is different

Reply:
Removed.